src/helpers.ts

```diff
@@ -1,9 +1,83 @@
 /**
  * Internal helper functions not intended to be published.
  */
-
 import {arraysEqual, extendedEuclid, gcd} from 'xen-dev-utils';
 
+// Lexicographically compare arrays; return true if a1 <= a2 lexicographically.
+// a1 < a2 if a1[i] < a2[i] where i is the first index where the two arrays differ...
+// ... or if a1 is a strict prefix of a2.// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+export function lexCompareArrays<T>(a1: T[], a2: T[]): boolean {
+  const [len1, len2] = [a1.length, a2.length];
+  for (let i = 0; i < Math.min(len1, len2); i++) {
+    if (a1[i] < a2[i]) {
+      return true;
+    } else if (a1[i] > a2[i]) {
+      return false;
+    }
+  }
+  // Now we know one is a prefix of the other.
+  if (len1 <= len2) {
+    return true;
+  } else {
+    return false;
+  }
+}
+/**
+ * Distribute subsequences as evenly as possible using Björklund's algorithm;
+ * modified as to always return the brightest mode.
+ */
+export function bjorklund_brightest_bool(a: number, b: number): boolean[] {
+  if (isNaN(a) || isNaN(b)) {
+    throw new Error('Invalid input');
+  }
+  const d = gcd(a, b);
+  if (d === 1) {
+    let [countFirst, countSecond] = [a / d, b / d];
+    // `false` is brighter; `false` < `true` in js.
+    let first = [false];
+    let second = [true];
+    while (countSecond !== 1) {
+      // After switching, are there more copies of `first` than `second`?
+      // Then all the `second`s go below the first `countSecond` copies of `first`s.
+      if (countFirst > countSecond) {
+        [countFirst, countSecond] = [countSecond, countFirst - countSecond];
+        [first, second] = [first.concat(second), first];
+      }
+      // Otherwise, there are strictly fewer `first`s than `second`s,
+      // and all the `first`s get modified, whereas `second` is unchanged.
+      // `countFirst` is unchanged.
+      else {
+        countSecond = countSecond - countFirst;
+        first = first.concat(second);
+      }
+      // At the current step we have `countFirst` `first` substrings and `countSecond` `second` substrings,
+      // where we must guarantee that `first` is lexicographically greater than `second`.
+      // Thus if first > second, then switch them and the count variables.
+      // Do this step before checking the while condition; we know the desired lex. ordering holds for the first step,
+      // and our stopping condition requires that `first` < `second` actually hold to really behave correctly.
+      if (!lexCompareArrays<boolean>(first, second)) {
```

Comment:
TypeScript is fairly smart with inference. I don't think explicit `<boolean>` is required here.